HW06.Rmd

```diff
@@ -61,28 +83,74 @@ detach(package:EnsDb.Hsapiens.v75, unload = T)
 detach(package:ensembldb, unload = T)
 
 #add in your own gene of interest!!! 
-genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2")
+#genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2")
 
+# LIN28 is my gene of interest and I have added that at the end of this vector. 
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2", "LIN28A")
+
+# In the GBM.genes.of.interest file that I obtained, I only got 27 observations instead of 28. Upon further inspection, I realized that the gene PTEN was repeated twice in the previous vector and that explained my result. I was initially worried!
 
 GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
 
 #Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
 
-#print out the t-test results
-```
+# I initially tried the following codes but I kept encountering an error with the filter function. 
+
+#for (gene in genes_of_interest) {
+#  WT <- GBM.genes.of.interest %>%
+#  filter(gene_symbol == gene, type == "WT")
+  
+#  mutated <- GBM.genes.of.interest %>%
+#  filter(gene_symbol == gene, type == "H3K27M_mutated")
+#}
 
+# I decided to go a different route and reorganize the table first this time. 
 
+reorg.GBM.genes.of.interest <- GBM.genes.of.interest %>% 
+  gather(key = "key", value = "value", -gene_symbol)  %>%
+  separate(key, into = c ("group", "order"), convert = TRUE)
+
+# Now that I have the table reorganized, I created a for loop to fo through each gene and run a ttest between mutated and WT.
+

```

Comment:
Yeah, I definitely think reorganizing the table is required before the for loop or filter functions will work. I also tried just going directly into the for loop for the t test but ran into errors. I think I'll try something similar to what you've done here with filtering after the reorganization to try and get my t test working.

Reply:
I think I had a similar issue to Ben, where I was trying to go right into the t-test instead of reorganizing the data.  This looks similar to what I was using to create my graphs so if I did this before the t-test like you did I probably would have had more success.